### PR TITLE
rolling.quantile returns an interpolated result

### LINKED ISSUE #9413: BUG: pandas rolling_quantile does not use interpolation
i recently bumped an unexpected issue with pandas rolling funcs. rolling_quantile for example:

```
>> row = 10
>> col = 5
>> idx = pd.date_range(20100101,periods=row,freq='B')
>> a = pd.DataFrame(np.random.rand(row*col).reshape((row,-1)),index=idx)
>> a
                   0           1           2           3           4
2010-01-01  0.341434    0.497274    0.596341    0.259909    0.872207
2010-01-04  0.222653    0.056723    0.064019    0.936307    0.785647
2010-01-05  0.179067    0.647165    0.931266    0.557698    0.713282
2010-01-06  0.049766    0.259756    0.945736    0.380948    0.282667
2010-01-07  0.385036    0.517609    0.575958    0.050758    0.850735
2010-01-08  0.628169    0.510453    0.325973    0.263361    0.444959
2010-01-11  0.099133    0.976571    0.602235    0.181185    0.506316
2010-01-12  0.987344    0.902289    0.080000    0.254695    0.753325
2010-01-13  0.759198    0.014548    0.139858    0.822900    0.251972
2010-01-14  0.404149    0.349788    0.038714    0.280568    0.197865

>> a.quantile([0.25,0.5,0.75],axis=0)
               0           1           2           3           4
0.25    0.189963    0.282264    0.094964    0.255999    0.323240
0.50    0.363235    0.503864    0.450966    0.271964    0.609799
0.75    0.572164    0.614776    0.600761    0.513510    0.777567

>> np.percentile(a,[25,50,75],axis=0)
[array([ 0.18996316,  0.28226404,  0.09496441,  0.25599853,  0.32323997]),
 array([ 0.36323529,  0.50386356,  0.45096554,  0.27196429,  0.60979881]),
 array([ 0.57216415,  0.61477607,  0.6007611 ,  0.51351021,  0.7775667 ])]

>> pd.rolling_quantile(a,row,0.25).tail(1)
                   0           1       2           3           4
2010-01-14  0.179067    0.259756    0.08    0.254695    0.282667
```

looks like pandas.DataFrame.quantile member func is consistent with the numpy.percentile func. however the pandas.rolling_quantile func returns diff results. reduce the row number to 5, the problem will be gone (all three methods return the same results). any thoughts?

ps: i also tested rolling_std func which will "randomly" generate errors with 10^-7 ~ 10^-8 scales (compared to pandas.DataFrame std member func or numpy/scipy std funcs which could limit the error close to np.spacing(1) level) for long (row-wise) pandas.DataFrames

python environment:

python 3.4.2
cython 0.21.1
numpy 1.8.2
scipy 0.14.0
pandas 0.15.1
statsmodels 0.6.0

### LINKED ISSUE #16211: rolling.quantile does not behave like quantile and np.percentile.
#### Code Sample

```python
>>> import numpy as np
>>> import pandas as pd
>>> arr = np.arange(100)
>>> s = pd.Series(arr)
>>> q1 = s.quantile(0.1)
>>> q2 = s.rolling(100).quantile(0.1).iloc[-1]
>>> q3 = s.quantile(0.1, interpolation='lower')
>>> q4 = s.rolling(100).apply(np.percentile, args=(10,)).iloc[-1]
>>> q5 = s.rolling(100).apply(np.percentile, args=(10, None, None, False, 'lower')).iloc[-1]
>>> np.isclose(q2, q1)
False
>>> np.isclose(q2, q3)
True
>>> np.isclose(q2, q4)
False
>>> np.isclose(q2, q5)
True
```
#### Problem description

Current implementation of quantile as a method of the rolling window behaves like the lower interpolation of Series.quantile and np.percentile. It would be nice if the default behavior was the same.

#### Expected Output

```python
>>> import numpy as np
>>> import pandas as pd
>>> arr = np.arange(100)
>>> s = pd.Series(arr)
>>> q1 = s.quantile(0.1)
>>> q2 = s.rolling(100).quantile(0.1).iloc[-1]
>>> q3 = s.quantile(0.1, interpolation='lower')
>>> q4 = s.rolling(100).apply(np.percentile, args=(10,)).iloc[-1]
>>> q5 = s.rolling(100).apply(np.percentile, args=(10, None, None, False, 'lower')).iloc[-1]
>>> np.isclose(q2, q1)
True
>>> np.isclose(q2, q3)
False
>>> np.isclose(q2, q4)
True
>>> np.isclose(q2, q5)
False
```

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.0.final.0
python-bits: 64
OS: Linux
OS-release: 4.10.8-1-ARCH
machine: x86_64
processor: 
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.19.2
nose: None
pip: 9.0.1
setuptools: 34.4.1
Cython: 0.25.2
numpy: 1.12.1
scipy: 0.19.0
statsmodels: None
xarray: None
IPython: 5.3.0
sphinx: 1.5.4
patsy: None
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: None
tables: 3.3.0
numexpr: 2.6.2
matplotlib: 2.0.0
openpyxl: 2.4.5
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.3
bs4: None
html5lib: 0.999999999
httplib2: None
apiclient: None
sqlalchemy: 1.1.9
pymysql: None
psycopg2: 2.7.1 (dt dec pq3 ext lo64)
jinja2: 2.9.6
boto: None
pandas_datareader: 0.3.0.post

</details>

### PR DESCRIPTION
**Pull request by guillemborrell**
- [x] closes #9413 
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/test_window.py [hunk 1]
```diff
@@ -1079,7 +1079,18 @@ def scoreatpercentile(a, per):
...
+
+            if int(idx) == values.shape[0] - 1:
+                retval = values[-1]
+
+            else:
+                qlow = int(idx) / (values.shape[0] - 1)
+                qhig = (int(idx) + 1) / (values.shape[0] - 1)
+                vlow = values[int(idx)]
+                vhig = values[int(idx + 1)]
+                retval = vlow + (vhig - vlow)*(per - qlow)/(qhig - qlow)
```
Comment (jreback) [inline pandas/tests/test_window.py]: this will cause some lint issues

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a release note (0.20.1), call a bug fix is fine. 
pls add the example from the issue as an additional tests; add a comment to reference the issue

**Comment by jreback** [comment]
add the examples as tests from #9413

**Comment by guillemborrell** [comment]
In my last master. The linter passed OK, and I have created a whatsnew entry in the Numeric section.

**Comment by guillemborrell** [comment]
@jreback  I think that I have implemented all the requested improvements.

**Comment by guillemborrell** [comment]
@jreback. Tests passed and changes implemented.

**Comment by guillemborrell** [comment]
@jreback I closed this a created a clean PR: #16247

**Comment by jreback** [comment]
pls don't open new pr's for the same issue, you simply push to the same PR. ok on this one.

**Comment by guillemborrell** [comment]
Sorry. Just trying to make things right.

**Pull request closed without merging**